src/windows/cheats.c

```diff
@@ -499,6 +565,81 @@ static void window_cheats_misc_dropdown(rct_window *w, int widgetIndex, int drop
 	game_do_command(0, GAME_COMMAND_FLAG_APPLY, CHEAT_FORCEWEATHER, dropdownIndex, GAME_COMMAND_CHEAT, 0, 0);
 }
 
+enum
+{
+	CONVERT_NUMBER_TO_STRING,
+	CONVERT_STRING_TO_NUMBER,
+};
+
+static void window_cheats_convert_money(int convertMethod, char* text) 
```

Comment:
This whole function is incredibly complex for what it does. I'd much rather see a `money_to_string` and `string_to_money` conversion methods, rather than this. They could consume a pointer to whatever string buffer is required.
As for string to money conversion: why can't you just find decimal separator, `atoi` everything you can to the left of it, then to the right of it? Consider using `sscanf` for it, as it returns number of bytes/chars consumed.